java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -199,7 +171,7 @@ public <RequestT, ResponseT, ErrorT> CompletableFuture<ResponseT> performRequest
         try {
             OpenSearchRequestBodyBuffer requestBody = prepareRequestBody(request, endpoint, options);
             SdkHttpFullRequest clientReq = prepareRequest(request, endpoint, options, requestBody);
-            if (asyncHttpClient != null) {
+            if (this.httpClient instanceof SdkAsyncHttpClient) {
                 return executeAsync(clientReq, requestBody, endpoint, options);
```

Comment:
We could probably pass the client by argument and eliminate the casting inside `executeAsync`:

```suggestion
                return executeAsync((SdkAsyncHttpClient)this.httpClient, clientReq, requestBody, endpoint, options);
```